clients/redigo.go

```diff
@@ -0,0 +1,389 @@
+}
+
+// JSONSet used to set a json object
+//
+// ReJSON syntax:
+// 	JSON.SET <key> <path> <json>
+// 			 [NX | XX]
+//
+func (r *Redigo) JSONSet(key string, path string, obj interface{}, opts ...rjs.SetOption) (
+	res interface{}, err error) {
+
+	if len(opts) > 1 {
+		return nil, rjs.TooManyOptionals
+	}
+	args := make([]interface{}, 0, 5)
+	args = append(args, key, path, obj)
+
+	if len(opts) == 1 {
+		args = append(args, opts[0].Value()...)
+	}
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_SET, args...)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONGet used to get a json object
+//
+// ReJSON syntax:
+// 	JSON.GET <key>
+//			[INDENT indentation-string]
+//			[NEWLINE line-break-string]
+//			[SPACE space-string]
+//			[NOESCAPE]
+//			[path ...]
+//
+func (r *Redigo) JSONGet(key, path string, opts ...rjs.GetOption) (res interface{}, err error) {
+
+	if len(opts) > 4 {
+		return nil, rjs.TooManyOptionals
+	}
+	args := make([]interface{}, 0)
+	args = append(args, key)
+
+	args = append(args, path)
+
+	for _, op := range opts {
+		args = append(args, op.Value()...)
+	}
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_GET, args...)
+	if err != nil {
+		return nil, err
+	}
+
+	return r.Conn.Do(name, args...)
+}
+
+// JSONMGet used to get path values from multiple keys
+//
+// ReJSON syntax:
+// 	JSON.MGET <key> [key ...] <path>
+//
+func (r *Redigo) JSONMGet(path string, keys ...string) (res interface{}, err error) {
+
+	if len(keys) == 0 {
+		return nil, rjs.NeedAtleastOneArg
+	}
+	args := make([]interface{}, 0)
+	for _, key := range keys {
+		args = append(args, key)
+	}
+
+	args = append(args, path)
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_MGET, args...)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONDel to delete a json object
+//
+// ReJSON syntax:
+// 	JSON.DEL <key> <path>
+//
+func (r *Redigo) JSONDel(key string, path string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_DEL, key, path)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONType to get the type of key or member at path.
+//
+// ReJSON syntax:
+// 	JSON.TYPE <key> [path]
+//
+func (r *Redigo) JSONType(key, path string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_TYPE, key, path)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONNumIncrBy to increment a number by provided amount
+//
+// ReJSON syntax:
+// 	JSON.NUMINCRBY <key> <path> <number>
+//
+func (r *Redigo) JSONNumIncrBy(key, path string, number int) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_NUMINCRBY, key, path, number)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONNumMultBy to increment a number by provided amount
```

Comment:
Seems to be a copy paste from JSONNumIncrBy. Might want to reword this to "... multiply a number ..."

Reply:
Done